### ISSUE TITLE
Linux Version? :pray: 

### ISSUE THREAD

**Issue by vexersa** [Rem0o/FanControl.Releases#535]
I would love to know if there are any plans to make a Linux compatible version of FanControl?

**Comment by Rem0o** [comment]
Nop, it's an old shcool .NET 4.7 app.

**Comment by vexersa** [comment]
Arg but it's so good.

**Comment by Rem0o** [comment]
Sorry.  If I ever modernize its stack there would be a possibility for Linux, but that would be a huge task still.

**Comment by MostHated** [comment]
Just throwing it out there that I +1 a Linux version.

**Comment by zeus86** [comment]
would love to see a linux version, too. it would be really awesome this project to become the platform-independent "OpenRGB" for just for fans.

**Comment by 8836290** [comment]
+1 for Linux please.

**Comment by Dradeliomecus** [comment]
+1 for Linux !!

**Comment by bob808** [comment]
+1 for Linux

**Comment by dongryphon** [comment]
I would pay $ for you to get a linux version if that would help... maybe other linux folks would also?

**Comment by vhouriet** [comment]
I would also be willing to contribute financially to this app if it would help create a potential Linux version

**Comment by Ryder17z** [comment]
+1 for Linux

**Comment by Rem0o** [comment]
What backend could be used on Linux?

Does https://github.com/LibreHardwareMonitor/LibreHardwareMonitor picks up fan controls ?

**Comment by zeus86** [comment]
> What backend could be used on Linux?
pretty sure best bet would be `lm-sensors`, which already does most of the heavy lifting...

**Comment by Rem0o** [comment]
One thing that bothers me is .NET 6.  I like 4.7/4.8 apps because that framework is built-in into windows and people don't have to download it and install it.  Whereas .NET Core++ apps either you bundle the whole damn framework with your app, for a whopping 100+ mb app, or you require the user to download the framework. Both options are awful imo.

**Comment by MostHated** [comment]
> One thing that bothers me is .NET 6. I like 4.7/4.8 apps because that framework is built-in into windows and people don't have to download it and install it. Whereas .NET Core++ apps either you bundle the whole damn framework with your app, for a whopping 100+ mb app, or you require the user to download the framework. Both options are awful imo.

Maybe it's just me, but I would gladly accept a 100mb app size if it meant I finally got a much-needed and sought-after tool. I would also venture a guess that many folks who are using Linux and want/need a tool like this would have absolutely no issue installing .Net in order to have it, or already have it installed.

**Comment by mtabbakh** [comment]
A Linux version, please!

**Comment by nick987** [comment]
I would definitely be keen to financially contribute to a project like this if a linux version was created! No programming skills whatsoever but I can vote with my wallet!

**Comment by mail2onur** [comment]
A Linux GUI version built on top of lm-sensors would be awesome

**Comment by dongryphon** [comment]
there is a fancontrol package for accessing the fans but I have not had tons of success configuring it ... hence my interest :) I would imagine one could make a daemon process that the GUI talks to so the policies it defines could be enforced w/o the (large) GUI running. Not sure if you are thinking to port the .NET app or move it to a new stack so just throwing ideas out there

**Comment by scchow** [comment]
+1 would definitely be willing to contribute code /test to help work towards a Linux version. Current solutions available on Linux are so much more limited and clunky compared to what you have here.

**Comment by Rem0o** [comment]
Since I guess lm-sensors is a process, how would it look if we query it for let say 30 sensors every second? Will each call spawn a new process?

**Comment by Inshal240** [comment]
Donated for potential linux support. Hope that helps!

**Comment by foursixnine** [comment]
@Rem0o Hey! Wanna work together trying to see if it works with Mono framework on linux? I guess that by the recent exposure from Jayz you might get a considerable amount of requests, and I'm sure a lot of us can help :)

**Comment by SophiaHadash** [comment]
I came here from JayZ only to find out it doesn't support linux... I really hope there will be a linux version in the future

**Comment by Rem0o** [comment]
@SophiaHadash 
It would be basically a completely different software. Software that plays with hardware like that is really OS specific.  I ain't a linux user myself, so not much incentive to work on such a new project.

**Comment by foursixnine** [comment]
Hey @Rem0o See my comment above, perhaps there's something that can be done? I do know that lmsensors could do a good job, but I'm not sure how your software works in windows :) cc @scherenhaenden

**Comment by Rem0o** [comment]
Did some preliminary testing for fun:

1) LHM does have a basic lm-sensors readout, but no fan control. However, with a bit of code, managed to get manual fan control for all detected PWM in ubuntu. Surprisingly easy to do so. Must be root user though.
2) FanControl is a WPF app, and WPF works in dotnet 6, but on windows only. Which means, would have to create a whole new UI project with an other UI tech stack. That's a whole new project by itself. Also, AFAIK, there is no mature dotnet GUI framework on Linux right now worth investing time in.


TLDR: FanControl isn't directly portable to Linux. Most of the logic could, but the app structure, UI and everything else would need to be started over from scratch.

**Comment by mail2onur** [comment]
> TLDR: FanControl isn't directly portable to Linux. Most of the logic could, but the app structure, UI and everything else would need to be started over from scratch.


Yes. I wished that you would have created Fancontrol on a cross platform toolkit when i first saw the application. That was the first thing came to my mind when i saw how greatly useful the application

**Comment by CZEMacLeod** [comment]
> 1. LHM does have a basic lm-sensors readout, but no fan control. However, with a bit of code, managed to get manual fan control for all detected PWM in ubuntu. Surprisingly easy to do so. Must be root user though.
> 2. FanControl is a WPF app, and WPF works in dotnet 6, but on windows only. Which means, would have to create a whole new UI project with an other UI tech stack. That's a whole new project by itself. Also, AFAIK, there is no mature dotnet GUI framework on Linux right now worth investing time in.
> 
> TLDR: FanControl isn't directly portable to Linux. Most of the logic could, but the app structure, UI and everything else would need to be started over from scratch.

I have to admit, I would quite like to see a version of this that runs as a service in the background on windows and linux, with a config/monitor UI separate. 
The UI could either be WPF on windows, or perhaps rewritten as blazor so it could be web or app based. I think that would make this already excellent tool just that little bit better. It would also make it (more) suitable for headless devices, servers and such too, which would also be great!

**Comment by vexersa** [comment]
> > 1. LHM does have a basic lm-sensors readout, but no fan control. However, with a bit of code, managed to get manual fan control for all detected PWM in ubuntu. Surprisingly easy to do so. Must be root user though.
> > 2. FanControl is a WPF app, and WPF works in dotnet 6, but on windows only. Which means, would have to create a whole new UI project with an other UI tech stack. That's a whole new project by itself. Also, AFAIK, there is no mature dotnet GUI framework on Linux right now worth investing time in.
> > 
> > TLDR: FanControl isn't directly portable to Linux. Most of the logic could, but the app structure, UI and everything else would need to be started over from scratch.
> 
> I have to admit, I would quite like to see a version of this that runs as a service in the background on windows and linux, with a config/monitor UI separate. The UI could either be WPF on windows, or perhaps rewritten as blazor so it could be web or app based. I think that would make this already excellent tool just that little bit better. It would also make it (more) suitable for headless devices, servers and such too, which would also be great!

This is an awesome idea!

**Comment by zeus86** [comment]
> I have to admit, I would quite like to see a version of this that runs as a service in the background on windows and linux, with a config/monitor UI separate. The UI could either be WPF on windows, or perhaps rewritten as blazor so it could be web or app based. I think that would make this already excellent tool just that little bit better. It would also make it (more) suitable for headless devices, servers and such too, which would also be great!

...this would also enable native frontends for different flavors (gtk/qt5) especially including an ncurses-interface, which would enable users to use the backend on the shell and thus not limiting the functionality to Instances with gut but also servers and headless-machines and such.

**Comment by Quatshcop** [comment]
I guess a good UI Framework would be Avalonia (avaloniaui.net). It's essentially WPF but Cross Platform.
The lack of this app on linux is one of the major blockers for me to completely move over. xD

Thank you so much for your work!

**Comment by mail2onur** [comment]
> 



> 

> 


> 

Ohh please don't do that. No one wants to download 200 mb file and install yet another dot.net framework. Could you please skip dot.net all together if/when you overhaul your app engine? Imo dot.net is even worse than Java based apps

**Comment by Rem0o** [comment]
@mail2onur  Well, it does uses .NET, 4.8 right now to be exact. I'm based on LHM, so I can't escape .NET. But 4.X is legacy at this point, and will have to be phased out at some point.

**Comment by bartimaeusnek** [comment]
> With .NET6, either I have to ship the framework with the app (portable), for a whooping 180 mb + app, or I rely on every user to download the framework and install the framework.

From .NET Core 1 onwards, the app itself will ask the user to install the framework by providing a download link in a window that opens itself. It should be no problem for Users to interact with it and install the framework.

**Comment by Rem0o** [comment]
@bartimaeusnek yup saw that, but it's not a direct download link, you still have to choose like x86, x64 and so on, and I can guarantee that at least 25% of users will do it wrong.

**Comment by Ryder17z** [comment]
Related, but it only lists up to version 4.8
https://docs.microsoft.com/en-us/dotnet/framework/deployment/deployment-guide-for-developers

You should be able to construct the same for newer versions but that seems to lack documentation so i don't know if it's still doable.

**Comment by SourTrigger** [comment]
> @bartimaeusnek yup saw that, but it's not a direct download link, you still have to choose like x86, x64 and so on, and I can guarantee that at least 25% of users will do it wrong.

I genuinely don't believe that this is true. I would wager that those people couldn't even figure out how to download a release off of Github either, nor would they likely be inclined to even care about fan curves or temperatures beyond stock functionality. If I were to draw a Venn diagram, I bet the circles would overlap 99%. I think you're fine excluding it from the app and having users download it directly from Microsoft's site.

**Comment by d249u7** [comment]
+1 for Linux. Even a headless "engine" would go a long way and I'd be willing to pay for linux support as well.

**Comment by jrentsch** [comment]
Another +1 for Lunix. Love this on Windows, Linux would be awesome.

**Comment by icavalheiro** [comment]
@Rem0o have you considered the upcoming .net version's native compilation? I know it's precisely build for this kind of scenario, smaller file sizes, instant start ups, ....

I would also like to contribute to the development iof you ever decide to open up the source code.

**Comment by Rem0o** [comment]
@icavalheiro I have checked it out, but it has so many limitations, that porting an app that has any user interface to native compilation is basically impossible.

https://learn.microsoft.com/en-us/dotnet/core/deploying/trimming/incompatibilities

**Comment by wiiznokes** [comment]
Hi, I made an open source alternative [here](https://github.com/wiiznokes/LibreFanControl) for those interested. Works also on Linux.

**Comment by san616mofo** [comment]
@Rem0o, what about Flutter (as a graphical interface)? I have a system with 13 fans (including two liquid cooling ones), and I would like to have full control over them on Linux. Looking forward to this version.

**Comment by nijahplays** [comment]
+1 for Linux :)

**Comment by DaRacci** [comment]
+1 for Linux

**Comment by funny-family** [comment]
+1 for Linux

**Comment by Navya-Rawat999** [comment]
+1 for linux
:)

**Comment by Ryder17z** [comment]
Previously mentioned repo is replaced by https://github.com/wiiznokes/fan-control

**Comment by Zyarra** [comment]
+1 :)
You could open a "kickstarter"-like thing for the project for an amount that would worth your time investment. Pretty sure it wouldnt take long to reach it.

**Comment by qtqgyt** [comment]
...Would Mono work?
+1

**Comment by thegame4craft** [comment]
> ...Would Mono work?
> +1

No as fancontrol uses WPF (Windows Presentation Framework) which is not existant in mono/linux

**Comment by Anexgohan** [comment]
just adding my +1 for linux version.
lm-sensors works pretty well for monitoring as well as pwmconfig

my output for lm-sensors:
```
sensors
```

```
it8628-isa-0a40
Adapter: ISA adapter
in0:           1.10 V  (min =  +0.00 V, max =  +3.06 V)
in1:           2.03 V  (min =  +0.00 V, max =  +3.06 V)
in2:           2.05 V  (min =  +0.00 V, max =  +3.06 V)
in3:           2.02 V  (min =  +0.00 V, max =  +3.06 V)
in4:           1.10 V  (min =  +0.00 V, max =  +3.06 V)
in5:         912.00 mV (min =  +0.00 V, max =  +3.06 V)
in6:           1.38 V  (min =  +0.00 V, max =  +3.06 V)
3VSB:          3.31 V  (min =  +0.00 V, max =  +6.12 V)
Vbat:          3.10 V
fan1:         686 RPM  (min =    0 RPM)
fan2:        3013 RPM  (min =    0 RPM)
fan3:         454 RPM  (min =    0 RPM)
fan4:           0 RPM  (min =    0 RPM)
fan5:         680 RPM  (min =    0 RPM)
temp1:        +37.0°C  (low  = +127.0°C, high = +127.0°C)  sensor = thermistor
temp2:        +37.0°C  (low  =  +0.0°C, high = +127.0°C)  sensor = thermistor
temp3:        +54.0°C  (low  =  +0.0°C, high = +127.0°C)
temp4:        +42.0°C
temp5:        +46.0°C
temp6:        +48.0°C
intrusion0:  OK

nvme-pci-0400
Adapter: PCI adapter
Composite:    +39.9°C  (low  = -273.1°C, high = +69.8°C)
                       (crit = +79.8°C)
Sensor 1:     +40.9°C  (low  = -273.1°C, high = +65261.8°C)
Sensor 2:     +39.9°C  (low  = -273.1°C, high = +65261.8°C)
Sensor 5:     +54.9°C  (low  = -273.1°C, high = +65261.8°C)

acpitz-acpi-0
Adapter: ACPI interface
temp1:        +16.8°C
temp2:        +16.8°C
temp3:        +16.8°C

k10temp-pci-00c3
Adapter: PCI adapter
Tctl:         +55.5°C
Tccd1:        +58.2°C
Tccd2:        +51.2°C

gigabyte_wmi-virtual-0
Adapter: Virtual device
temp1:        +37.0°C
temp2:        +37.0°C
temp3:        +55.0°C
temp4:        +42.0°C
temp5:        +45.0°C
temp6:        +48.0°C
```

# -----
fan1:         686 RPM  (min =    0 RPM)   <-- fan1 is CPU_FAN     - CPU Front Fan
fan2:        3013 RPM  (min =    0 RPM)   <-- fan2 is SYS_FAN1    - Top Fan
fan3:         454 RPM  (min =    0 RPM)   <-- fan3 is SYS_FAN2    - Back Fan
fan4:           0 RPM  (min =    0 RPM)   <-- fan4 is PCH_FAN     - PCH Fan
fan5:         680 RPM  (min =    0 RPM)   <-- fan5 is CPU_OPT     - CPU Rear Fan

**Comment by xX-MrN0b0dy-Xx** [comment]
+1 for Linux

**Comment by DaRacci** [comment]
id recommend using this project for linux if anyone is still searching.

https://gitlab.com/coolercontrol/coolercontrol

**Comment by xX-MrN0b0dy-Xx** [comment]
> id recommend using this project for linux if anyone is still searching.
> 
> https://gitlab.com/coolercontrol/coolercontrol

Yes, I saw it in the thread, thank you!

It was just to say that Id love to see a Linux version too

**Comment by xQuantoM** [comment]
1+ for LINUX

**Comment by GabrielSalla** [comment]
+1 for Linux

**Comment by Benzeel** [comment]
+1 for linux version！ still one of the things keeping me from switching over because of how heavily i rely on fancontrol

**Comment by ExileFox** [comment]
> +1 for linux version！ still one of the things keeping me from switching over because of how heavily i rely on fancontrol

https://gitlab.com/coolercontrol/coolercontrol might suffice instead

**Comment by qtqgyt** [comment]
> Sorry. If I ever modernize its stack there would be a possibility for Linux, but that would be a huge task still.

This is .NET correct? With the proper APIs we could just use Mono

**Comment by Rem0o** [comment]
@qtqgyt  not for the whole UI which is WPF, which uses DirectX.

**Comment by Rem0o** [comment]
<img width="626" height="516" alt="Image" src="https://github.com/user-attachments/assets/96d58d1b-562b-491f-ad25-549dfd50b818" />

Took a shot at it with a dirty proto, as the architecture is pretty modular now.  
With a combination of this: https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/2355

I can run the full Fancontrol "engine" with a (limited) LHM backend.  It works.  However hardware support from Windows to Linux is not transferable at all, so no GPU support at the moment, but Linux being Linux, my guess is that it shouldn't be too hard to bring in the minimum requirement to make it usable.

Also, configuration files from Windows to Linux would not work, as sensor mapping from Windows to Linux is not the same. 

But yeah, started to play with it.

**Comment by mail2onur** [comment]
> Took a shot at it with a dirty proto, as the architecture is pretty modular now. With a combination of this: [LibreHardwareMonitor/LibreHardwareMonitor#2355](https://github.com/LibreHardwareMonitor/LibreHardwareMonitor/pull/2355)
> 
> I can run the full Fancontrol "engine" with a (limited) LHM backend. It works. However hardware support from Windows to Linux is not transferable at all, so no GPU support at the moment, but Linux being Linux, my guess is that it shouldn't be too hard to bring in the minimum requirement to make it usable.
> 
> Also, configuration files from Windows to Linux would not work, as sensor mapping from Windows to Linux is not the same.
> 
> But yeah, started to play with it.

Wow, it's finally coming after 4 years. I have deleted Windows in the meantime and using Cooler Control app at CachyOS but i would gladly switch to good old Fan Control when it's ready

**Comment by viyullas** [comment]
Glad to hear about this.
It would be great, now thay I am moving to Arch, to be able to bring FanControl with me.

**Comment by Anexgohan** [comment]
I was one of the requesters for Linux support originally, but ended up making my own fan control. It has native Linux and Windows support and is capable of support multi platform and more than 1 machine simultaneously.
Check it out 
https://github.com/Anexgohan/pankha
